common/src/main/java/org/apache/sedona/common/raster/serde/Serde.java

```diff
@@ -176,4 +178,19 @@ public static GridCoverage2D deserialize(byte[] bytes) throws IOException, Class
             return state.restore();
         }
     }
+
+    public static byte[] serializeGridSampleDimension(GridSampleDimension sampleDimension) {
```

Comment:
This was already implemented in Sedona. Why implement it again?

Reply:
This is as per Kristin's recommendation to implement new serialization/deserialization methods for GridSampleDimension that use `kryos.get()` to retrieve `Kryo` objects. `kryos.get()` avoids repeatedly initializing new Kryo instances.
I should have used the existing `GridSampleDimensionSerializer` for read and write into output buffer. This is added now.